src/TSHelper.ts

```diff
@@ -484,7 +484,10 @@ export class TSHelper {
             const objType = this.inferAssignedType(expression.parent.parent, checker);
             const property = objType.getProperty(expression.parent.name.getText());
             if (!property) {
-                return objType.getStringIndexType();
+                const stringPropertyType = objType.getStringIndexType();
+                if (stringPropertyType) {
+                    return stringPropertyType;
+                }
```

Comment:
Isn't this equivalent to what it was before? Or does it return false

Reply:
This ensures that if there is no string index type, it falls back to the default, which is to return the type of the original expression itself (see end of function)